email-ballerina/imap_listener_endpoint.bal

```diff
@@ -0,0 +1,164 @@
+
+    }
+
+    # Stops the `email:ImapListener` forcefully.
+    # ```ballerina
+    # email:Error? result = emailListener.immediateStop();
+    # ```
+    #
+    # + return - `()` or else a `email:Error` upon failure to stop the listener
+    public isolated function immediateStop() returns error? {
+        check self.stop();
+    }
+
+    # Stops the `email:ImapListener` gracefully.
+    # ```ballerina
+    # email:Error? result = emailListener.gracefulStop();
+    # ```
+    #
+    # + return - () or else error upon failure to stop the listener
+    public isolated function gracefulStop() returns error? {
+        check self.stop();
+    }
+
+    isolated function internalStart() returns error? {
+        var scheduler = self.config.cronExpression;
+        if (scheduler is string) {
+            task:AppointmentConfiguration config = {cronExpression: scheduler};
+            self.appointment = new(config);
+        } else {
+            task:TimerConfiguration config = {intervalInMillis: self.config.pollingInterval, initialDelayInMillis: 100};
+            self.appointment = new (config);
+        }
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.attach(imapAppointmentService, self);
+            check appointment.start();
+        }
+        log:print("User " + self.config.username + " is listening to remote server at " + self.config.host + "...");
+    }
+
+    isolated function stop() returns error? {
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.stop();
+        }
+        log:print("Stopped listening to remote server at " + self.config.host);
+    }
+
+    isolated function poll() returns error? {
+        return poll(self);
+    }
+
+    # Registers for the Email service.
+    # ```ballerina
+    # emailListener.register(helloService, hello);
+    # ```
+    #
+    # + emailService - Type descriptor of the service
+    # + name - Service name
+    public isolated function register(service object {} emailService, string? name) {
+        register(self, emailService);
+    }
+}
+
+final service isolated object{} imapAppointmentService = service object {
+    remote isolated function onTrigger(ImapListener l) {
+        var result = l.poll();
+        if (result is error) {
+            log:printError("Error while executing poll function", result);
+        }
+    }
+};
+
+# Configuration for Email listener endpoint.
+#
+# + host - Email server host
+# + username - Email server access username
+# + password - Email server access password
+# + protocolConfig - IMAP4 protocol configuration
+# + pollingInterval - Periodic time interval to check new update
+# + cronExpression - Cron expression to check new update
+public type ImapListenerConfig record {|
+    string host;
+    string username;
+    string password;
+    ImapConfig? protocolConfig = ();
```

Comment:
Don't have to have this inside this anymore right ?

Reply:
Replaced with its inner fields.